errbot/flow.py

```diff
@@ -122,7 +122,7 @@ class Flow(object):
     This is a live Flow. It keeps context of the conversation (requestor and context).
     Context is just a python dictionary representing the state of the conversation.
     """
-    def __init__(self, root: FlowRoot, requestor: Identifier, initial_context: Mapping[str, Any]):
+    def __init__(self, root: FlowRoot, requestor: Identifier, initial_context: Mapping[str, Any], dismiss_prompting=False):
```

Comment:
We're also trying to have [type hints](https://www.python.org/dev/peps/pep-0484/) on all our (at least user-facing) APIs, so we should define this as such:

```python
def __init__(self, root: FlowRoot, requestor: Identifier, initial_context: Mapping[str, Any],
                    next_step_hinting: int=True):
```

We should also document the parameter in the docstring below. Something like:

`:param next_step_hinting: toggle display of the "you are now in the flow xxx, you may continue with yyy, zzz" messages after each flow step`